### PR TITLE
feat(spanner/spansql): support Table rename & Table synonym

### LINKED ISSUE #9226: spanner: Allow `RENAME TABLE` and `ALTER TABLE ... RENAME TO` in the parser
**Is your feature request related to a problem? Please describe.**

The go client currently rejects `RENAME TABLE ... TO` and `ALTER TABLE ... RENAME TO`, but Spanner will soon support the syntax.
In addition, `SYNONYM` will also be supported for graceful table renaming. This adds DDL statements such as `CREATE TABLE (... SYNONYM (name))`, `ALTER TABLE ... ADD SYNONYM ...`, and `ALTER TABLE ... DROP SYNONYM ...`.

**Describe the solution you'd like**
Allow `RENAME TABLE ... TO`, `ALTER TABLE ... RENAME TO`, `CREATE TABLE (... SYNONYM (name))`, `ALTER TABLE ... ADD SYNONYM ...`, and `ALTER TABLE ... DROP SYNONYM ...` in the ddl parser.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by killah777**
fixes #9226

### CODE REVIEW

File: spanner/spansql/parser.go [hunk 1]
```diff
@@ -982,7 +982,7 @@ func (p *parser) parseDDLStmt() (DDLStmt, *parseError) {
 
 	/*
 		statement:
-			{ create_database | create_table | create_index | alter_table | drop_table | drop_index | create_change_stream | alter_change_stream | drop_change_stream }
+			{ create_database | create_table | create_index | alter_table | drop_table | rename_table | drop_index | create_change_st     ream | alter_change_stream | drop_change_stream }
```
Comment (rahul2393) [inline spanner/spansql/parser.go:985]:
```suggestion
			{ create_database | create_table | create_index | alter_table | drop_table | rename_table | drop_index | create_change_stream | alter_change_stream | drop_change_stream }
```

Comment (killah777) [reply spanner/spansql/parser.go]: Thank you! Did not realize the mistake. First time on github or open-source.

**Comment by rahul2393** [comment]
Thanks @killah777, Will be merging this once feature is available for public use.

**Review by rahul2393 — APPROVED** [review]

**Comment by killah777** [comment]
Hello, the official launch date is today, and the feature is publicly available. (User documentation to be pushed today)
Can this be merged? Thanks!

**Pull request merged**